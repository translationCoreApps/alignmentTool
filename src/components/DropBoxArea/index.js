import React, {Component} from 'react';
import PropTypes from 'prop-types';
// components
import DropBoxItem from '../DropBoxItem';

class DropBoxArea extends Component {

  render() {
    const {contextIdReducer, wordAlignmentReducer, } = this.props;
    let { bibles } = this.props.resourcesReducer;
    if (contextIdReducer.contextId) {
      let { chapter, verse } = contextIdReducer.contextId.reference;
      let alignments = wordAlignmentReducer.alignmentData[chapter][verse].alignments;
      let verseText = bibles['ugnt'][chapter] ? bibles['ugnt'][chapter][verse] : null;
      return (
        <div style={{ display: 'flex', flexWrap: 'wrap', height: '100%', backgroundColor: '#ffffff', padding: '0px 10px 50px', overflowY: 'auto' }}>
        {
          alignments.map((alignment, index) => {
            return (
              <DropBoxItem
                key={index}
                alignmentIndex={index}
                bottomWords={alignment.bottomWords}
                topWords={alignment.topWords}
                onDrop={item => this.handleDrop(index, item)}
                verseText={verseText}
                actions={this.props.actions}
                resourcesReducer={this.props.resourcesReducer}
              />
            );
          })
        }
        </div>
      );
    }
    return <div />;
  }

  handleDrop(index, wordBankItem) {
    this.props.actions.moveWordBankItemToAlignment(index, wordBankItem);
  }
}

DropBoxArea.propTypes = {
  wordAlignmentReducer: PropTypes.object.isRequired,
  contextIdReducer: PropTypes.object.isRequired,
  actions: PropTypes.object.isRequired
};

export default DropBoxArea;
