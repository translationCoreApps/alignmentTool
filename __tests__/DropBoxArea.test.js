import React from 'react';
import DropBoxArea from '../src/components/DropBoxArea';
import renderer from 'react-test-renderer';

test('DropBoxArea renders', () => {
    const component = renderer.create(
        <DropBoxArea />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // TODO: exercise UI
});