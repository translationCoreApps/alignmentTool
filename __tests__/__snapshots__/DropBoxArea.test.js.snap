// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropBoxArea renders 1`] = `<DropBoxArea />`;
